src/realm/util/file_mapper.cpp

```diff
@@ -576,12 +531,7 @@ EncryptedFileMapping* add_mapping(void* addr, size_t size, const FileAttributes&
     }
     catch (...) {
         if (it->info->mappings.empty()) {
-#ifdef _WIN32
-            bool b = CloseHandle(it->info->fd);
-            REALM_ASSERT_RELEASE(b);
-#else
-            ::close(it->info->fd);
-#endif
+            File::close_static(it->info->fd);
```

Comment:
since now it may throw on failure, may be first erase then close?

Reply:
Good point, updated.